lib/fetchers/local.rb

```diff
@@ -39,12 +47,39 @@ def self.resolve_from_string(target)
       target if File.exist?(target)
     end
 
-    def initialize(target)
+    def initialize(target, opts = {})
       @target = target
+      @do_vendor = opts[:do_vendor]
+      @backend = opts[:backend]
+      @archive_shasum = nil
     end
 
-    def fetch(_path)
-      archive_path
+    def fetch(path)
+      return @target unless @do_vendor
+
+      # Skip vendoring if @backend is not set (example: ad hoc runners)
+      return @target unless @backend
+
+      if File.directory?(@target)
+        # Create an archive, checksum, and move to the vendor directory
+        Dir.mktmpdir do |tmpdir|
+          final_path = File.join(tmpdir, "#{File.basename(@target)}.tar.gz")
+          opts = {
+            backend: @backend,
+            output: final_path,
+          }
+          Inspec::Profile.for_target(@target, opts).archive(opts)
+          checksum = perform_shasum(final_path)
+          FileUtils.mv(final_path, File.join(path, "#{checksum}.tar.gz"))
+        end
+      else
+        # Verify profile (archive) is valid and extract to vendor directory
+        opts = { backend: @backend }
+        Inspec::Profile.for_target(@target, opts).check
+        Inspec::FileProvider.for_path(@target).extract(path)
```

Comment:
Just a question here. It seems odd that if its a directory we tar it up and store it in the cache. If it is not a directory (an archive) we extract it. Should we just copy the archive over to keep consistency?

Reply:
Sure, I don't see any harm in that. I'll make it happen.

Reply:
Thinking of this more its fine. We will never have the case where we will have a local .tar file from this fetcher. This path is only used when vendoring and vendoring will extract everything anyways downstream.